src/shared/randombytes.c

```diff
@@ -59,7 +59,7 @@ void randombytes(void *ptr, size_t length)
     static int fh = -1;
     ssize_t ret;
 
-    if (fh < 0 && (fh = open("/dev/urandom", O_RDONLY), fh < 0 && (fh = open("/dev/random", O_RDONLY), fh < 0))) {
+    if (fh < 0 && (fh = open("/dev/urandom", O_RDONLY | __O_CLOEXEC), fh < 0 && (fh = open("/dev/random", O_RDONLY | __O_CLOEXEC), fh < 0))) {
```

Comment:
Could we replace `__O_CLOEXEC` with `O_CLOEXEC`? We should avoid system private definitions.